### PR TITLE
 Extension name collision in the extension list.

### PR DESCRIPTION
**Pull request by infokub**
New behavior : 

When a name collide in the Extension list there is a new case : 
* If the name collides and the origin exists in the local extension, and equals the Asset Store tag, then nothing, it behaves like before
* If there are no origin set, or another tag, the Install button becomes "Override", the tag "Already installed" becomes "Already in project" and an other alert popup explains what happen

### CODE REVIEW

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionInstallDialog.js [hunk 1]
```diff
@@ -127,17 +133,21 @@ const ExtensionInstallDialog = ({
...
+					  extensionShortHeader.tier === 'community' ? (
+						<Trans>Update (could break the project)</Trans>
+					  ) : (
+						<Trans>Update</Trans>
+					  )
+					) : (
+					  <Trans>Re-install</Trans>
+					)
+				) : (
+					<Trans>Override</Trans>
```
Comment (4ian) [inline newIDE/app/src/AssetStore/ExtensionStore/ExtensionInstallDialog.js:147]: We could go for a simple worder I think: "Replace existing extension"?

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js [hunk 2]
```diff
@@ -81,7 +84,7 @@ export const ExtensionListItem = ({
               {alreadyInstalled && (
                 <Chip
                   size="small"
-                  label={<Trans>Already installed</Trans>}
+                  label={fromStore ? <Trans>Already installed</Trans> : <Trans>Already in project</Trans>}
```
Comment (4ian) [inline newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js:87]: Do you think we should use the word "conflict" here? To show that things are coming from a "different background"?

```suggestion
                  label={fromStore ? <Trans>Already installed</Trans> : <Trans>Conflict with extension in project</Trans>}
```

**Review by 4ian — COMMENTED** [review]
Overall, looks good to me! Added some minor comments.

To have the CI checks to pass, just run `npm run format` in newIDE/app. This will automatically format the source code.
Then, it should be good to go 👍

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js (hunk 2 above)
Comment (infokub) [reply newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js:87]: We can't bc of compatibility issues with older projects.
Origin didn't exist so i didn't want to break the wording for them with something as scary as a conflict can be.
So i chose a smooth and unconvincing (or even confusing...) wording.

Tbh i don't have many ideas about it. A wording fitting for
- someonse who really downloaded the extension, but on an old project
- someone who only made his extension with a conflicting name

is not easy 😄 

Source : https://github.com/4ian/GDevelop/blob/master/Core/GDCore/Project/EventsFunctionsExtension.h#L167

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionInstallDialog.js (hunk 1 above)
Comment (infokub) [reply newIDE/app/src/AssetStore/ExtensionStore/ExtensionInstallDialog.js:147]: Thanks, that's right, i didn't use format, and ok for the Override change !

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js (hunk 2 above)
Comment (4ian) [reply newIDE/app/src/AssetStore/ExtensionStore/ExtensionListItem.js]: You're very right, great investigation. A wording that is less frightening is a good idea as for now we still have projects that had extensions coming from the store without the origin being properly registered.

**Comment by 4ian** [comment]
Sorry for the delay here, it totally slipped my mind. Looks great and thanks for the explanations. Merged 👍

**Pull request merged**